modules/kafka/kafka-dest-driver.c

```diff
@@ -236,20 +272,69 @@ _construct_client(KafkaDestDriver *self)
   if (!client)
     {
       msg_error("kafka: error constructing the kafka connection object",
-                evt_tag_str("topic", self->topic_name),
+                evt_tag_str("topic", self->topic_name->template),
                 evt_tag_str("error", errbuf),
                 evt_tag_str("driver", self->super.super.super.id),
                 log_pipe_location_tag(&self->super.super.super.super));
     }
   return client;
 }
 
+static gboolean
+_contains_valid_pattern(const gchar *name)
+{
+  const gchar *p;
+  for (p = name; *p; p++)
+    {
+      if (!((*p >= 'a' && *p <= 'z') ||
+            (*p >= 'A' && *p <= 'Z') ||
+            (*p >= '0' && *p <= '9') ||
+            (*p == '_') || (*p == '-') || (*p == '.')))
+        {
+          return FALSE;
+        }
+    }
+  return TRUE;
+}
+
+gboolean
+kafka_dd_validate_topic_name(const gchar *name)
+{
+  gint len = strlen(name);
+  if(len == 0)
+    {
+      msg_error("kafka: topic name is illegal, it can't be empty", evt_tag_str("topic", name));
+      return FALSE;
+    }
+  if((!g_strcmp0(name, ".")) || !g_strcmp0(name, ".."))
+    {
+      msg_error("kafka: topic name cannot be . or ..", evt_tag_str("topic", name));
+      return FALSE;
+    }
+  if(len > 249)
+    {
+      msg_error("kafka: topic name cannot be longer than 249 characters", evt_tag_str("topic", name), evt_tag_printf("length",
+                "%d", len));
+      return FALSE;
+    }
+  if(!_contains_valid_pattern(name))
+    {
+      msg_error("kafka: topic name is illegal as it contains characters other than pattern [\\-_a-zA-Z0-9]+",
```

Comment:
`.` is also accepted by `_contains_valid_pattern()`, please add it to the regexp here, or remove it from the check.